### PR TITLE
tweak popover css

### PR DESCRIPTION
**Pull request by keithamus**
_Authors: Please fill out this form carefully and completely._

_Reviewers: By approving this Pull Request you are approving the code change, as well as its deployment and mitigation plans._
_Please read this description carefully. If you feel there is anything unclear or missing, please ask for updates._

### Screenshots

N/A

#### List the issues that this change affects.

Fixes issues reported in https://github.com/orgs/community/discussions/52083#discussioncomment-5983786

#### Risk Assessment
  

- [x] **Low risk** the change is small, highly observable, and easily rolled back.
- [ ] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### What approach did you choose and why?

I've tweaked the CSS to account for cases where the browser supports `:open`, `:popover-open`, both, or neither.

The trade-off is slightly more CSS. The risk is that future iterations on the CSS may cause regressions (but that was true anyway). 

### Anything you want to highlight for special attention from reviewers?

I've tested this in some specific browser combinations, and what used to be buggy behaviour is now consistent:

 - Safari Technical Preview 166
 - Chrome 113 (with Experimental Web Platform Features enabled)
 - Seamonkey 2.53.16 (Firefox 91)
 - Waterfox Classic 2022.11
 - Qutebrowser 2.5.4 (Chromium 87)

I've also tested this with and without the upstream patches in https://github.com/oddbird/popover-polyfill/pull/103 and it works identically in both, which is a very good sign.

### Accessibility

- **No new axe scan violation** - This change does not introduce any new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violations.

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Lookbook)
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [x] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: app/components/primer/alpha/overlay.pcss [hunk 1]
```diff
@@ -3,13 +3,21 @@
...
-anchored-position {
-  display: block;
+@supports selector(:popover-open) {
+  .Overlay[popover]:not(.\:popover-open) {
+    display: revert;
+  }
+}
+
+@supports selector(:open) {
+  .Overlay[popover]:not(.\:popover-open) {
```
Comment (jonrohan) [inline app/components/primer/alpha/overlay.pcss:20]: Should this be?

```suggestion
  .Overlay[popover]:not(.\:open) {
```

Comment (keithamus) [reply app/components/primer/alpha/overlay.pcss:20]: It should not. This is specifically _reverting_ the `display` property, to revert lines 9-11

Comment (langermank) [reply app/components/primer/alpha/overlay.pcss:20]: It might be nice to add that as a comment above the declaration @keithamus just for more clarity :D

**Review by camertron — APPROVED** [review]

**Comment by camertron** [comment]
Hey @keithamus, I looked into the build failure just now and discovered it's happening because we're using an older version of autoprefixer in docs/. We can't upgrade because our old version of Gatsby restricts which version of autoprefixer we can install. I think the solution here is to disable autoprefixer for primer_view_components.css, since it's already been run through autoprefixer by the build script. I went ahead and made that change to this PR. Looks like the build is passing now.

**Comment by keithamus** [comment]
Thanks for tackling that @camertron!

**Review by camertron — APPROVED** [review]

**Pull request merged**